libraries/AP_NavEKF/AP_NavEKF_Source.h

```diff
@@ -51,6 +51,13 @@ class AP_NavEKF_Source
         FUSE_ALL_VELOCITIES = (1 << 0)  // fuse all velocities configured in source sets
     };
 
+    // enum for source set idx
+    enum SourceSetSwitch : uint8_t {
+        SOURCE_SET_SWITCH_PRIMARY = 0,
+        SOURCE_SET_SWITCH_SECONDARY = 1,
+        SOURCE_SET_SWITCH_TERTIARY = 2,
+    };
```

Comment:
```suggestion
    enum class SourceSet : uint8_t {
        PRIMARY = 0,
        SECONDARY = 1,
        TERTIARY = 2,
    };
```

and move it into the `AP_NavEKF_Source` class

Reply:
> I think it's already in the AP_NavEKF_Source class.
> SourceSet is defined here. so I used SourceSetSwitch.

Darn!  How about `SourceSetSelection`?  These sets are usually selected using a script rather than a switch.

> If we change it to enum class, we also have to change the functions to take this enum.

They probably already should've :-)

> The functions are defined in various places, and we have to change a lot for this.
> 
>     * AP_AHRS_NavEKF::set_posvelyaw_source_set
> 
>     * AP_AHRS - set_posvelyaw_source_set
> 
>     * AP_NavEKF_Source::setPosVelYawSourceSet
> 
>     * NavEKF3::setPosVelYawSourceSet
> 
>     * bindings.desc - set_posvelyaw_source_set
> 
> 
> Do you still think we need to change it to enum class?

If we're going to do it we may as well do it right....

Maybe we should do that in a separate PR so we can get the other two patches in?  Or get those other two into a different PR?

Reply:
OK, I will separate PR.
There is already a discussion of EKF_POS_SOURCE here, so I will create another PR of KILL_IMU.